### PR TITLE
[MRG] Added epilepsy example dataset

### PR DESCRIPTION
**Pull request by ftadel**

### CODE REVIEW

File: ieeg_epilepsy/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_channels.tsv [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+s'1	SEEG	uV	n/a	n/a	512	intracranial	s'	good
+s'2	SEEG	uV	n/a	n/a	512	intracranial	s'	good
+s'3	SEEG	uV	n/a	n/a	512	intracranial	s'	good
+s'10	SEEG	uV	n/a	n/a	512	intracranial	s'	good
+s'11	SEEG	uV	n/a	n/a	512	intracranial	s'	good
+s'12	SEEG	uV	n/a	n/a	512	intracranial	s'	good
+fz	EEG	uV	n/a	n/a	512	unknown		good
+cz	EEG	uV	n/a	n/a	512	unknown		good
+ecg1	ECG	uV	n/a	n/a	512	unknown		good
+ecg2	ECG	uV	n/a	n/a	512	unknown		good
```
Comment (sappelhoff) [inline ieeg_epilepsy/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_channels.tsv]: you are missing a column for the last 5 rows, if it's not known, fill it with `n/a`

File: ieeg_epilepsy/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_ieeg.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    "TaskName": "seizure",
+    "SamplingFrequency": 512,
+    "PowerLineFrequency": 50,
+    "SoftwareFilters", "n/a",
```
Comment (sappelhoff) [inline ieeg_epilepsy/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_ieeg.json]: typo: comma should be a colon

Comment (sappelhoff) [reply ieeg_epilepsy/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_ieeg.json]: (see also below for similar)

File: ieeg_epilepsy/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_channels.tsv (hunk 1 above)
Comment (sappelhoff) [reply ieeg_epilepsy/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_channels.tsv]: (see also below for similar)

**Comment by sappelhoff** [comment]
> Regarding the remaining warnings: should I add participants.json and _channels.json files?
Where can I find templates for these files? I couldn't find info about these in the specs or the article.

Yes, please specify a `participants.json` which contains a description of all "non-custom keys" that you are using in the corresponding `participants.tsv`. 

The "custom keys" are usually explicitly referred to (and described) in the specification ... such that additional information is not necessary. In your case `sex` throws an error, because there is no explanation about it in the specification, so you have to provide it. For more information on what's custom, you can see this file: https://github.com/bids-standard/bids-validator/blob/master/bids_validator/tsv/non_custom_columns.json

Regarding `channels.json` and `electrodes.json`, ... if you are using only the fields as defined in the specification, then you shouldn't add anything --> you are just getting an error, because the validator lags a bit behind the specification, and we have not yet added the "new custom columns" to the bids-validator. (This is a ToDo for us, cc @choldgraf )

Regarding a "template", please see: https://github.com/bids-standard/bids-examples/blob/bep006_eeg/eeg_matchingpennies/participants.json

**Comment by sappelhoff** [comment]
ahhhh, I completely forgot ... openneuro is running `bids-validator` ... but for now, to validate iEEG data, we need to run `bids-validator --bep010`.

This will only change once iEEG is merged into the main spec. Actually, we could already make that step with the validator IMHO, what's your take on this @chrisfilo? Would it be okay to remove the optional EEG and iEEG flags of the validator now? Or is that something we should do as the "last step of the merging process"?

**Comment by chrisfilo** [comment]
I would prefer to wait until it's merged into the spec. Hopefully won't be
long now.

**Comment by sappelhoff** [comment]
> I would prefer to wait until it's merged into the spec. Hopefully won't be long now.

Mmh okay fair enough, that puts us into a dilemma situation though: @ftadel's data is part of iEEG bids, which we want to publish before merging ... but in order to publish, it needs to be uploaded (his choice: openneuro), which is only possible once the merge has happened ;-)

Solutions I see:
- storing the data on another platform ... either temporarily or permanently
- not sharing the full data (yet) ... but waiting for after the merge --> which might be *before* the actual print of the iEEG publication, so only the people reading the preprint would be confused that the real example data is not accessible.

**Comment by ftadel** [comment]
When will this be merged?

**Comment by sappelhoff** [comment]
We are waiting for an approving review from @choldgraf or @dorahermes

It seems that in the meantime, adjustments to the validator also made this PR not pass:

```
Validating dataset ieeg_epilepsy/
	1: [ERR] Invalid JSON file. The file is not formatted according the schema. (code: 55 - JSON_SCHEMA_VALIDATION_ERROR)
		./sub-01/ses-post/ieeg/sub-01_ses-post_coordsystem.json
			Evidence:  should NOT have additional properties
	2: [ERR] Empty files not allowed. (code: 99 - EMPTY_FILE)
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_ieeg.eeg
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_ieeg.vhdr
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_ieeg.vmrk
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-02_ieeg.eeg
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-02_ieeg.vhdr
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-02_ieeg.vmrk
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-03_ieeg.eeg
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-03_ieeg.vhdr
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-03_ieeg.vmrk
	1: [WARN] Tabular file contains custom columns not described in a data dictionary (code: 82 - CUSTOM_COLUMN_WITHOUT_DESCRIPTION)
		./sub-01/ses-post/ieeg/sub-01_ses-post_space-MNI152_electrodes.tsv
			Evidence: Columns: name, x, y, z, size, hemisphere, group, type, manufacturer not defined, please define in: /space-MNI152_electrodes.json, /electrodes.json,/sub-01/sub-01_space-MNI152_electrodes.json,/sub-01/sub-01_electrodes.json,/sub-01/ses-post/sub-01_ses-post_space-MNI152_electrodes.json,/sub-01/ses-post/sub-01_ses-post_electrodes.json,/sub-01/ses-post/ieeg/sub-01_ses-post_space-MNI152_electrodes.json,/sub-01/ses-post/ieeg/sub-01_ses-post_electrodes.json
		./sub-01/ses-post/ieeg/sub-01_ses-post_space-orig_electrodes.tsv
			Evidence: Columns: name, x, y, z, size, hemisphere, group, type, manufacturer not defined, please define in: /space-orig_electrodes.json, /electrodes.json,/sub-01/sub-01_space-orig_electrodes.json,/sub-01/sub-01_electrodes.json,/sub-01/ses-post/sub-01_ses-post_space-orig_electrodes.json,/sub-01/ses-post/sub-01_ses-post_electrodes.json,/sub-01/ses-post/ieeg/sub-01_ses-post_space-orig_electrodes.json,/sub-01/ses-post/ieeg/sub-01_ses-post_electrodes.json
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_channels.tsv
			Evidence: Columns: reference, group not defined, please define in: /sub-01/ses-post/ieeg/sub-01_ses-post_channels.json, /sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_channels.json,/sub-01/ses-post/sub-01_ses-post_channels.json,/sub-01/ses-post/sub-01_ses-post_task-seizure_channels.json,/sub-01/sub-01_channels.json,/sub-01/sub-01_task-seizure_channels.json,/channels.json,/task-seizure_channels.json,/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-01_channels.json
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-02_channels.tsv
			Evidence: Columns: reference, group not defined, please define in: /sub-01/ses-post/ieeg/sub-01_ses-post_channels.json, /sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_channels.json,/sub-01/ses-post/sub-01_ses-post_channels.json,/sub-01/ses-post/sub-01_ses-post_task-seizure_channels.json,/sub-01/sub-01_channels.json,/sub-01/sub-01_task-seizure_channels.json,/channels.json,/task-seizure_channels.json,/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-02_channels.json
		./sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-03_channels.tsv
			Evidence: Columns: reference, group not defined, please define in: /sub-01/ses-post/ieeg/sub-01_ses-post_channels.json, /sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_channels.json,/sub-01/ses-post/sub-01_ses-post_channels.json,/sub-01/ses-post/sub-01_ses-post_task-seizure_channels.json,/sub-01/sub-01_channels.json,/sub-01/sub-01_task-seizure_channels.json,/channels.json,/task-seizure_channels.json,/sub-01/ses-post/ieeg/sub-01_ses-post_task-seizure_run-03_channels.json
        Summary:                 Available Tasks:        Available Modalities: 
        42 Files, 33.17KB        seizure                 T1w                   
        1 - Subject                                      ieeg                  
        2 - Sessions                                                           
If you have any questions please post on https://neurostars.org/tags/bids
```

**Comment by choldgraf** [comment]
sorry, all of my cycles had gone towards getting the BIDS-iEEG paper submitted. I can try to take a stab at finishing up the validator code. Then we should re-run on this PR and see what modifications need to be made to ensure this dataset passes the validator

**Comment by choldgraf** [comment]
Just a note that updates are now merged into the validator, so we should see how it works on this dataset once more. Looking at those errors:

* The invalid JSON file error should be fixed now
* The "empty files" errors can be fixed by just writing in some text into those files (e.g. `with open('myfile.vhdr', 'w') as ff: ff.write('some random text')`. It's a bit hacky but I think it should work
* The "undocumented columns" warning will be smaller (since we now expect x, y, z, size, etc) though you'll need to provide a JSON file to explain the remaining columns.

**Comment by ftadel** [comment]
Passed!

**Comment by sappelhoff** [comment]
Thanks @ftadel ! :-)

**Pull request merged**